article/article.md

```diff
@@ -0,0 +1,153 @@
+
+When changing the tiles but maintaining the colorization rules, we could interpolate between the floating-point values of each pixel in the old and new tiles, then compute the color of the interpolated value. This would mean each pixel would transition along the linear gradient(s) defined for the color scale.
+
+```glsl
+// retrieve and decode pixel value from the old tile
+vec4 rgbaFloatsA = texture2D(textureA, vTexCoordA);
+float pixelFloatValueA = rgbaToFloat(rgbaFloatsA, littleEndian);
+
+// retrieve and decode pixel value from the new tile
+vec4 rgbaFloatsB = texture2D(textureB, vTexCoordB);
+float pixelFloatValueB = rgbaToFloat(rgbaFloatsB, littleEndian);
+
+// interpolate
+float interpolated = mix(pixelFloatValueA, pixelFloatValueB, interpolationFraction);
+
+// compute the color of the interpolated value
+gl_FragColor = computeColor(interpolated, colorScale, sentinelValues, colorScaleLength, sentinelValuesLength);
+```
+
+When changing both tiles and colorization rules, though, this approach wouldn't work, because old and new tiles needed to be colorized according to different rules. In this case, we'd need to compute the color of each pixel in the old tiles and each pixel in the new tiles and then interpolate (in RGB color space) between the _colors_.
+
+```glsl
+// ... obtain old and new pixel values as before
+
+// compute the color of the old pixel
+vec4 colorA = computeColor(pixelFloatValueA, colorScaleA, sentinelValuesA, colorScaleLengthA, sentinelValuesLengthA);
+
+// compute the color of the new pixel
+vec4 colorB = computeColor(pixelFloatValueB, colorScaleB, sentinelValuesB, colorScaleLengthB, sentinelValuesLengthB);
+
+// interpolate between the colors
+gl_FragColor = mix(colorA, colorB, interpolationFraction);
+```
+
+Note that I've simplified the shader code here for the purpose of highlighting the basic concepts. For curious readers, here are the actual implementations for [interpolating by value](https://github.com/ihmeuw/leaflet.tilelayer.glcolorscale/blob/master/src/shaders/interpolateValue.frag.glsl) and [interpolating by color](https://github.com/ihmeuw/leaflet.tilelayer.glcolorscale/blob/master/src/shaders/interpolateColor.frag.glsl).
+
+## Outcome
+
+The new approach to fetching and rendering map tiles in the LBD application yielded encouraging results in terms of performance and usability. Tile requests now complete much faster - generally in less than a second, compared with 3+ seconds in the prior implementation. This makes using the tool in geospatial (i.e. raster) mode a smoother experience, eliminating the frustration of having to wait a long time for map tiles to load. Seeing pixel values on mouse hover allows users immediate and continuous access to the estimates being visualized. And animated per-pixel transitions make the changes between one view and another easier to perceive, particularly helpful for visualizing changes over time.
```

Comment:
Ugh, don't start sentences with "And". "Finally, animated per-pixel transitions..."